mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,951 @@
+                        ])
+                    else:
+                        anno_box[new_idx] = torch.cat([
+                            ct - torch.tensor([x, y], device=device),
+                            torch.unsqueeze(z, dim=0), task_boxes[idx][k][3:6],
+                            torch.unsqueeze(vx, dim=0),
+                            torch.unsqueeze(vy, dim=0),
+                            torch.unsqueeze(torch.sin(rot), dim=0),
+                            torch.unsqueeze(torch.cos(rot), dim=0)
+                        ],
+                                                      dim=0)
+
+            hms.append(hm)
+            anno_boxes.append(anno_box)
+            masks.append(mask)
+            inds.append(ind)
+
+        return hms, anno_boxes, inds, masks
+
+    def loss(self, gt_bboxes_3d, gt_labels_3d, preds_dicts, **kwargs):
+        """Loss function for CenterHead.
+
+        Args:
+            gt_bboxes_3d (list[:obj:`LiDARInstance3DBoxes`]): Ground
+                truth gt boxes.
+            gt_labels_3d (list[torch.Tensor]): Labels of boxes.
+            preds_dicts (dict): Output of forward function.
+        """
+        hms, anno_boxes, inds, masks = self.get_targets(
+            gt_bboxes_3d, gt_labels_3d)
+        rets = []
+        for task_id, preds_dict in enumerate(preds_dicts):
+            # heatmap focal loss
+            preds_dict[0]['hm'] = self._sigmoid(preds_dict[0]['hm'])
+            hm_loss = self.crit(preds_dict[0]['hm'], hms[task_id])
+
+            target_box = anno_boxes[task_id]
+            # reconstruct the anno_box from multiple reg heads
+            if self.dataset == 'nuscenes':
+                preds_dict[0]['anno_box'] = torch.cat(
+                    (preds_dict[0]['reg'], preds_dict[0]['height'],
+                     preds_dict[0]['dim'], preds_dict[0]['vel'],
+                     preds_dict[0]['rot']),
+                    dim=1)
+            else:
+                raise NotImplementedError()
+
+            loss = 0
+            ret = {}
+
+            # Regression loss for dimension, offset, height, rotation
+            ind = inds[task_id]
+            pred = preds_dict[0]['anno_box'].permute(0, 2, 3, 1).contiguous()
+            pred = pred.view(pred.size(0), -1, pred.size(3))
+            pred = self._gather_feat(pred, ind)
+            mask = torch.unsqueeze(masks[task_id], -1).expand(pred.size())
+            box_loss = torch.sum(self.crit_reg(pred, target_box, mask), 1) / (
+                masks[task_id].float().sum() + 1e-4)
+
+            loc_loss = (box_loss *
+                        box_loss.new_tensor(self.code_weights)).sum()
+
+            loss += hm_loss + self.weight * loc_loss
+
+            ret.update({
+                'loss': loss,
+                'hm_loss': hm_loss.detach().cpu(),
+                'loc_loss': loc_loss,
+                'loc_loss_elem': box_loss.detach().cpu(),
+                'num_positive': masks[task_id].float().sum()
```

Comment:
Do we need `num_positive`, `loc_loss_elem`? Only keep standard keywords `loss_cls` `loss_bbox` should work, see `Anchor3DHead`.